snippets/header-drawer.liquid

```diff
@@ -55,7 +59,10 @@
                                   </a>
                                 {%- else -%}
                                   <details id="Details-menu-drawer-submenu-{{ forloop.index }}">
-                                    <summary class="menu-drawer__menu-item link link--text list-menu__item focus-inset">
+                                    <summary
+                                      id="HeaderMenu-{{ link.handle }}-{{ childlink.handle }}"
```

Comment:
I'm also not married to the dash separated format here. I considered doing `HeaderMenu_{{ link.handle }}_{{ childlink.handle }}` because the handles will already being using dashes. But I wasn't convinced there was a significant need to clearly indicate the separation.